src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsModel.cs

```diff
@@ -531,6 +524,119 @@ private void AddSchema(CsdlSchema schema, bool addAnnotations)
+                if (edmVersion == null || edmVersion < schema.Version)
+                {
+                    this.SetEdmVersion(schema.Version);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Process the <see cref="IEdmReference"/>s of the parent model to
+        /// figure out whether the schema's types (entiy types, enums, containers, etc.) and annotations should
+        /// be imported
+        /// </summary>
+        /// <param name="schema"></param>
+        /// <param name="schemaWrapper"></param>
+        /// <param name="parentReferences"></param>
+        /// <param name="shouldAddSchemaElements">Whether schema types such entity types, operations, enums, etc. should be important</param>
+        /// <param name="includeAnnotationsIndex">Cache of the schema's out-of-line <see cref="IEdmIncludeAnnotations"/>s indexed by the term namespace</param>
+        private static void ProcessSchemaParentReferences(
+            CsdlSchema schema,
+            CsdlSemanticsSchema schemaWrapper,
+            IEnumerable<IEdmReference> parentReferences,
+            out bool shouldAddSchemaElements,
+            out Dictionary<string, List<IEdmIncludeAnnotations>> includeAnnotationsIndex)
+        {
+            string schemaNamespace = schema.Namespace;
+
+            shouldAddSchemaElements = false;
+            includeAnnotationsIndex = new Dictionary<string, List<IEdmIncludeAnnotations>>();
+            
+
+            foreach (IEdmReference reference in parentReferences)
+            {
+                if (!shouldAddSchemaElements)
+                {
+                    // we should add types from this schema if there's at least
+                    // one edmx:Include that references this schema's namespace
+                    foreach (IEdmInclude include in reference.Includes)
+                    {
+                        if (include.Namespace == schemaNamespace)
+                        {
+                            shouldAddSchemaElements = true;
+                            break;
+                        }
+                    }
+                }
+
+                foreach(IEdmIncludeAnnotations includeAnnotations in reference.IncludeAnnotations)
+                {
+                    // index the edmx:IncludeAnnotations by namespace to make it
+                    // easier to filter which annotations to import with the schema
+                    string termNamespace = includeAnnotations.TermNamespace;
+                    List<IEdmIncludeAnnotations> cachedAnnotations;
+                    if (!includeAnnotationsIndex.TryGetValue(termNamespace, out cachedAnnotations))
+                    {
+                        cachedAnnotations = new List<IEdmIncludeAnnotations>();
+                        includeAnnotationsIndex[termNamespace] = cachedAnnotations;
+                    }
+
+                    cachedAnnotations.Add(includeAnnotations);

```

Comment:
After we add `includeAnnotations` to `cachedAnnotations` what do we do with that?

Reply:
Understood